src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -98,29 +104,125 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     LogMetricsFromAuthResult(authenticationResult, AuthenticationRequestParameters.RequestContext.Logger);
                     LogSuccessfulTelemetryToClient(authenticationResult, telemetryEventDetails, telemetryClients);
 
+                    ServiceBundle.PlatformProxy.OtelInstrumentation?.LogSuccessMetrics(
```

Comment:
Can this be moved into LogMsalSuccessTelemetryToOTel? This RunAsync method is getting a bit long, just thinking that extracting code into separate methods will make it more readable and maintainable.